rel/files/riak

```diff
@@ -1,57 +1,87 @@
+	    chown root:root $PID_DIR/riak.pid 2&>1 >/dev/null \
+                || true  # if we are run as non-root, just never mind
+	    break
+	fi
+	sleep 2
+	c=$(($c - 1))
+    done
+}
+
+function delete_pid_file {
+    rm -f $PID_DIR/riak.pid
+}
 
-# When running as a service, running as riak not as root, and systemd has created PID folder
-if [[ $EUID -ne 0 ]]; then
-    case "$1" in
-        start|console|foreground)
-            RELX_CONFIG_PATH=${RELX_CONFIG_PATH} VMARGS_PATH=${VMARGS_PATH} RUNNER_LOG_DIR=${RUNNER_LOG_DIR} PIPE_DIR=${PIPE_DIR} ${COMMAND} ${*} -pa {{platform_lib_dir}}/patches
-            ;;
-        *)
-            RELX_CONFIG_PATH=${RELX_CONFIG_PATH} VMARGS_PATH=${VMARGS_PATH} RUNNER_LOG_DIR=${RUNNER_LOG_DIR} PIPE_DIR=${PIPE_DIR} ${COMMAND} ${*}
-            ;;
-    esac
+# centos7-based distros have a su that contacts pam and prints the "Last logged in" message
+if [ "`cat /etc/redhat-release 2>&1`" = "CentOS Stream release 8" ] ||
+   [ "`cat /etc/system-release 2>&1`" = "Amazon Linux release 2 (Karoo)" ]; then
+    SU="su --session-command"
 else
-    # In this case we're running sudo riak - so have root access, but cannot rely
-    # systemd having created the PID dir, and need to sudo to the riak user
-    if [ ! -d $PID_DIR ]; then
-        mkdir $PID_DIR
-        chown riak:riak $PID_DIR
-    fi
-    case "$1" in
-        start|console|foreground)
-            su - riak -c "NODETOOL_NODE_PREFIX=${NODETOOL_NODE_PREFIX} RELX_CONFIG_PATH=${RELX_CONFIG_PATH} VMARGS_PATH=${VMARGS_PATH} RUNNER_LOG_DIR=${RUNNER_LOG_DIR} PIPE_DIR=${PIPE_DIR} ${COMMAND} ${*} -pa {{platform_lib_dir}}/patches"
-            ;;
-        debug)
-            # Drop the "debug" from the args as we're going to directly call riak-debug now
-            shift
-            # Debug may fail if run via relx script due to use of relative start location, and also need for root access
-            NODETOOL_NODE_PREFIX=${NODETOOL_NODE_PREFIX} RELX_CONFIG_PATH=${RELX_CONFIG_PATH} VMARGS_PATH=${VMARGS_PATH} RUNNER_LOG_DIR=${RUNNER_LOG_DIR} PIPE_DIR=${PIPE_DIR} ${DEBUG_COMMAND} ${*}
-            ;;
-        *)
-            ESCAPED_ARGS=`echo "$@" | sed -e 's/\([\\\(\\\){}"\x27]\)/\\\\\1/g'`
-            su - riak -c "NODETOOL_NODE_PREFIX=${NODETOOL_NODE_PREFIX} RELX_CONFIG_PATH=${RELX_CONFIG_PATH} VMARGS_PATH=${VMARGS_PATH} RUNNER_LOG_DIR=${RUNNER_LOG_DIR} PIPE_DIR=${PIPE_DIR} ${COMMAND} ${ESCAPED_ARGS}"
-            ;;
-    esac
+    SU=su
 fi
 
+function maybe_su {
+    if [[ $EUID -ne 0 ]]; then
+        $*
+    else
+	# if we are executing an admin command that spins up a
+	# (hidden) node to then execute custom erlang code via -eval,
+	# we need to cd to a dir containing the erlang cookie
+	# (previously implicitly done by su -, which option we have
+	# removed in order to allow any env vars to be available for
+	# the ultimate invocation of riak/riak-cs/stanchion)
+	cd {{platform_base_dir}}
+	$SU riak -c "$*"
+    fi
+}
 
+case "$1" in
```

Comment:
It's an effort to make hyphenated and non-hyphenated forms equivalent.

Reply:
I don't think it works as expected, and I'm not even sure that resurrecting the hyphenated commands is warranted - what purpose does it serve? It's been a couple of years since we moved to start script extensions. You now have to re-insert a bunch of boilerplate environment stuff at the top of the script files to make them work, whereas that was all inherited with from the root start script with extensions.

My opinion is that it's a backwards step, unless there's a compelling reason that I'm not aware of / considering.

The tidy up across the rest of the release config is very decent though, and much needed. Some of the deps currently reference TI forks, which would need to target basho, but guess that would be fixed up before everything gets merged in.

Reply:
To deal with the easier part first: re-pinning deps to TI-Tokyo repos has a single purpose: pull in cuttlefish and rebar3_cuttlefish, not so much for the fqdn data type in cuttlefish but rather for several commits in the plugin that make it compatible with rebar3 3.18 (by ioolkos, see [this one](https://github.com/TI-Tokyo/rebar3_cuttlefish/commit/57ef2c6bdb7b535d9423bfcde70a0057d8a5a73e) and thereabout), [synchronize](https://github.com/basho/rebar3_cuttlefish/commit/31d14b67fe8cc1173f11667873ea1b617db5ea06) its internal install_update_escript with the ["upstream" version](https://github.com/erlware/relx/blob/main/priv/templates/install_upgrade_escript) of it in relx, and last but not least, this [little gem](https://github.com/basho/rebar3_cuttlefish/commit/2c052177b7ba3e18664ae1b309d78342de9f19d1).

I note that the fqdn-type patch has already been [merged](https://github.com/basho/cuttlefish/commit/95a96b35e533e803cdcfc5dfd09dd308697840c9) in basho/cuttlefish, but there is no tag. If you do the tagging, I will then prepare the necessary PRs to bump cuttlefish version where necessary (it is an explicit dependency in bitcask, clique, eleveldb and riak_sysmon), and collect a patchset for another PR in rebar3_cuttlefish.

Regarding the phasing out of hyphenated forms, I have no strong opinions and will gladly follow the party line (although I would invite @Bob-The-Marauder to this discussion, who has been in the business of dealing with customers for some time).

For the record, my understanding of the underlying machinery is as follows:

* The `riak` script generated by relx and/or cuttlefish provides basic commands like start, stop, console, pid, ping, rpc. It is how riak is started/stopped in riak_test setting. It is not under our control.
* The other `riak`, the one created from the template in rel/files, has a distinct and specific purpose:
  - it takes care of the pid file (relx-generated one will not do this for us);
  - it does `su` as appropriate, and is therefore installed in PATH when riak is installed from a package;
  - it launches `riak-admin` etc when called as `riak admin`.

I saw this last item precisely as a way to enable non-hyphenated forms (if we remove riak-admin and its hyphenated friends from PATH, the move to hide hyphenated forms will be complete, from the user standpoint). Apparently, by "move to start script extensions" you mean to do it differently -- perhaps by injecting code in the script produced by relx? If so, I would be happy to rework my contribution accordingly, but I didn't see how relx can be massaged to achieve that.